go.mod

```diff
@@ -2,8 +2,13 @@ module github.com/hashicorp/terraform
 
 require (
 	cloud.google.com/go v0.36.0
-	github.com/Azure/azure-sdk-for-go v21.3.0+incompatible
-	github.com/Azure/go-autorest v10.15.4+incompatible
+	github.com/Azure/azure-sdk-for-go v31.2.0+incompatible
+	github.com/Azure/go-autorest v12.3.0+incompatible // indirect
+	github.com/Azure/go-autorest/autorest v0.5.0
```

Comment:
You probably don't need dependency on both 

```
	github.com/Azure/go-autorest v12.3.0+incompatible // indirect
	github.com/Azure/go-autorest/autorest v0.5.0
```

Please see here: https://github.com/Azure/go-autorest#using-with-go-modules

Reply:
@tamalsasha the project does not build without this. Without the root-level dependency the build process will attempt to fall back to an older version that causes conflicts.

You can try it yourself with a clean repo and then just attempting to update `github.com/Azure/azure-sdk-for-go` and then running a `go mod tidy`.

Reply:
Even doing `go get github.com/Azure/azure-sdk-for-go@v31.2.0 github.com/Azure/go-autorest@v12.3.0` followed by `go mod tidy` will attempt to add a bad version back, so you need to re-add `v12.3.0` afterwards.

Reply:
I didn't see the replace line though, so I'll give that a shot too!

Reply:
@tamalsasha yeah thanks for this. The big thing was the `replace` line for sure which I will get added to this PR.

Thanks!

Reply:
Glad it worked!